cwms-data-api/src/test/java/cwms/cda/data/dao/timeseriesprofile/TimeSeriesProfileParserDaoIT.java

```diff
@@ -0,0 +1,194 @@
+                timeSeriesProfileParserDao.deleteTimeSeriesProfileParser(profileParser.getLocationId().getName(), profileParser.getKeyParameter(), profileParser.getLocationId().getOfficeId());
+            }
+
+            timeSeriesProfileDao.deleteTimeSeriesProfile(timeSeriesProfileDepth.getLocationId().getName(), timeSeriesProfileDepth.getKeyParameter(),
+                    timeSeriesProfileDepth.getLocationId().getOfficeId());
+            timeSeriesProfileDao.deleteTimeSeriesProfile(timeSeriesProfileTemp.getLocationId().getName(), timeSeriesProfileTemp.getKeyParameter(),
+                    timeSeriesProfileTemp.getLocationId().getOfficeId());
+            assertEquals(2, profileParserList.size());
+        });
+    }
+
+    @Test
+    void testStoreAndCopy() throws SQLException {
+        CwmsDatabaseContainer<?> databaseLink = CwmsDataApiSetupCallback.getDatabaseLink();
+        databaseLink.connection(c -> {
+            DSLContext context = getDslContext(c, databaseLink.getOfficeId());
+
+            TimeSeriesProfileDao timeSeriesProfileDao = new TimeSeriesProfileDao(context);
+
+            TimeSeriesProfile timeSeriesProfile = buildTestTimeSeriesProfile("AAA", "Depth");
+            timeSeriesProfileDao.storeTimeSeriesProfile(timeSeriesProfile, false);
+
+            timeSeriesProfile = buildTestTimeSeriesProfile("BBB", "Depth");
+            timeSeriesProfileDao.storeTimeSeriesProfile(timeSeriesProfile, false);
+
+            TimeSeriesProfileParserDao timeSeriesProfileParserDao = new TimeSeriesProfileParserDao(context);
+            TimeSeriesProfileParser timeSeriesProfileParser = buildTestTimeSeriesProfileParser("AAA", "Depth");
+            timeSeriesProfileParserDao.storeTimeSeriesProfileParser(timeSeriesProfileParser, false);
+
+            TimeSeriesProfileParser retrieved = timeSeriesProfileParserDao.retrieveTimeSeriesProfileParser(timeSeriesProfileParser.getLocationId().getName(),
+                    timeSeriesProfileParser.getKeyParameter(), timeSeriesProfileParser.getLocationId().getOfficeId());
+
+            timeSeriesProfileParserDao.copyTimeSeriesProfileParser(timeSeriesProfileParser.getLocationId().getName(),
+                    timeSeriesProfileParser.getKeyParameter(), timeSeriesProfileParser.getLocationId().getOfficeId(), "BBB");
+
+            TimeSeriesProfileParser retrieved1 = timeSeriesProfileParserDao.retrieveTimeSeriesProfileParser("BBB",
+                    timeSeriesProfileParser.getKeyParameter(), timeSeriesProfileParser.getLocationId().getOfficeId());
+            assertEquals(timeSeriesProfileParser.getKeyParameter(), retrieved.getKeyParameter());
+            assertEquals("BBB", retrieved1.getLocationId().getName());
+            assertEquals(timeSeriesProfileParser.getTimeFormat(), retrieved1.getTimeFormat());
+
+        });
+    }
+
+    private static TimeSeriesProfile buildTestTimeSeriesProfile(String location, String parameter) {
+        String officeId = "HEC";
```

Comment:
Use a real office, add it to the user.sql in the test resources.

Reply:
now when using getWebUser, I have access to a real office. Switched to LRL